### PR TITLE
 feat(ripple): add feature targeting to remaining public mixins

### PR DESCRIPTION
**Pull request by mmalerba**
depends on #4308

### CODE REVIEW

File: test/scss/feature-targeting.scss [hunk 1]
```diff
@@ -4,35 +4,54 @@
 @import "@material/feature-targeting/functions";
 @import "@material/menu/mixins";
 @import "@material/radio/mixins";
+@import "@material/ripple/mixins";
 @import "@material/typography/mixins";
 
 // Button
-
-@include mdc-button($query: mdc-feature-any());
+.mdc-button {
```
Comment (kfranqueiro) [inline test/scss/feature-targeting.scss]: I know we talked about this elsewhere, but looking at this, it occurs to me that it's incorrect (and thus misleading if anyone were to take this as reference, although they shouldn't) to invoke a component's main `mdc-foo` mixin within a selector. (This only goes for components, e.g. not elevation/typography)

If we're doing this per-component, perhaps we should reserve that treatment only for individual public mixins, not the main one capable of emitting all the base styles (which would emit this selector itself)?

Alternatively we could just have a `.test` selector around the entire set...

File: packages/mdc-ripple/_mixins.scss [hunk 2]
```diff
@@ -89,9 +91,7 @@
   &::before {
     @include mdc-feature-targets($feat-animation) {
       // Also transition background-color to avoid unnatural color flashes when toggling activated/selected state
-      transition:
-        opacity $mdc-states-wash-duration linear,
-        background-color $mdc-states-wash-duration linear;
+      transition: opacity $mdc-states-wash-duration linear, background-color $mdc-states-wash-duration linear;
```
[the remark this answers is not in the document]
Comment (mmalerba) [reply packages/mdc-ripple/_mixins.scss]: Yep... it's really nice for making quick changes without having to worry about getting the indentation right. I just wish it handled a few things differently

File: test/scss/feature-targeting.scss (hunk 1 above)
Comment (mmalerba) [reply test/scss/feature-targeting.scss]: yeah that's a good point, I'll just put `.test` around everything

Comment (mmalerba) [reply test/scss/feature-targeting.scss]: went with `.mdc-test` in the end due to the linter

**Pull request merged**